pkg/controller/controller_broker.go

```diff
@@ -205,6 +206,14 @@ func (c *controller) reconcileBroker(broker *v1alpha1.Broker) error {
 		}
 		glog.V(5).Infof("Successfully converted catalog payload from Broker %v to service-catalog API", broker.Name)
 
+		if len(catalog) == 0 {
+			s := fmt.Sprintf("Error getting catalog payload for broker %q; received zero services; at least one service is required", broker.Name)
+			glog.Warning(s)
+			c.recorder.Eventf(broker, api.EventTypeWarning, errorSyncingCatalogReason, s)
+			c.updateBrokerCondition(broker, v1alpha1.BrokerConditionReady, v1alpha1.ConditionFalse, errorSyncingCatalogReason, errorSyncingCatalogMessage+s)
+			return stderrors.New(s)
```

Comment:
now that I'm seeing this pattern over and over and over again. it seems like there should be a function that takes a string and does:
 - logging
 - event recorder
 - update condition
 - makes an error to return

all wrapped up into a function.

Food for thought.